workflow/controller/workflowpod.go

```diff
@@ -1015,18 +1016,27 @@ func (woc *wfOperationCtx) addArchiveLocation(tmpl *wfv1.Template) {
 	if !needLocation {
 		return
 	}
-	archiveLog := tmpl.ArchiveLocation.IsArchiveLogs()
+	archiveLog := woc.shouldArchiveLogs(tmpl)
 	tmpl.ArchiveLocation = woc.artifactRepository.ToArtifactLocation()
 	// overwrites archivelogs on/off with priorities:
-	// controller: on,  template: off -> result: on
-	// controller: on,  template: on  -> result: on
-	// controller: off, template: off -> result: off
-	// controller: off, template: on  -> result: on
+	// controller(on) > template > workflow > controller(off)
 	if archiveLog {
```

Comment:
Something about this is wrong, historically. What about a new environment variable we set on the wait container? `ARGO_ARCHIVE_LOGS`, that the controller sets. This would be very explicit and make it easy to debug issues.

Reply:
Instead of additional `ARGO_ARCHIVE_LOGS`, 
maybe we can capture all this logic in `woc.IsArchiveLogs`
I think this is explicit as well

I've added this change,
let me how it goes, can change to `ARGO_ARCHIVE_LOGS` if required